app/assets/javascripts/admin_legacy/views/promotional_features/_form.js

```diff
@@ -0,0 +1,48 @@
+(function () {
+  'use strict'
+  window.GOVUK = window.GOVUK || {}
+
+  var promotionalFeaturesForm = {
+    init: function init (options) {
+      this.form = document.querySelector(options.selector)
+      this.toggleImageAndYoutubeFieldsVisibility()
+      this.addRadioButtonEventListeners()
+    },
+
+    toggleImageAndYoutubeFieldsVisibility: function toggleImageAndYoutubeFieldsVisibility () {
+      var form = this.form
+      var imageRadioButtonChecked = form.querySelector('#promotional_feature_promotional_feature_items_attributes_0_image_or_youtube_video_url_image').checked
+      var youtubeRadioButtonChecked = form.querySelector('#promotional_feature_promotional_feature_items_attributes_0_image_or_youtube_video_url_youtube_video_url').checked
+
+      if (!imageRadioButtonChecked) {
+        var imageFields = form.querySelector('.image-fields')
+        imageFields.style.display = 'none'
+      }
+
+      if (!youtubeRadioButtonChecked) {
+        var youtubeFields = form.querySelector('.youtube-video-url-fields')
+        youtubeFields.style.display = 'none'
+      }
+    },
+
+    addRadioButtonEventListeners: function addRadioButtonEventListeners () {
+      var form = this.form
+      var imageRadioButton = form.querySelector('#promotional_feature_promotional_feature_items_attributes_0_image_or_youtube_video_url_image')
+      var youtubeRadioButton = form.querySelector('#promotional_feature_promotional_feature_items_attributes_0_image_or_youtube_video_url_youtube_video_url')
+      var imageFields = form.querySelector('.image-fields')
+      var youtubeFields = form.querySelector('.youtube-video-url-fields')
+
+      imageRadioButton.addEventListener('click', function () {
```

Comment:
I'm not 100% sure but I think monitoring click might not be as effective as monitoring change on radio buttons, the latter feels more idiomatic but I am rusty in this area.

You can also probably manage this with a single event e.g:
```
var radioElements = document.querySelector('<radio buttons'>)

radioElements.addEventListener('change', function (event) {
  if (event.value == 'youtube_video_url') {
    // show youtube hide images
  } else {
    // show images hide youtube

  }
})

```

Reply:
as it returns i had to use `querySelectorAll` and iterate over the results. Still simpler though for sure 👍